docs/source/modifying-messages.rst

```diff
@@ -0,0 +1,390 @@
+====================
+Message manipulation
+====================
+
+To create malformed messages tlsfuzzer supports modifying them before sending.
+You can apply the modifiers on generator nodes, the ones that send messages
+to the other side.
+
+Custom message generation
+=========================
+
+Tlsfuzzer provides support for sending arbitrary messages over established
+connections.
+It provides two nodes to achieve it: one to send message unencrypted and
+one to send it using current connection status.
+
+Creating unencrypted messages
+-----------------------------
+
+To send a record with a specific payload and type, irrespective of
+active encryption or negotiated fragmentation, use
+:py:class:`~tlsfuzzer.messages.PlaintextMessageGenerator`.
+It accepts two parameters to specify data sent to the other
+side (``content_type`` and ``data``) as well as one
+used for debugging: ``description``, printed when sending of the message
+failed.
+
+.. note::
+    As it skips all the usual message processing steps, it also doesn't
+    update handshake hashes so values calculated for Finished and connection
+    secrets in :term:`TLS` 1.3 won't match expected ones.
+
+For example, to send an empty ClientHello message, write:
+
+.. code:: python
+
+    node = node.add_child(PlaintextMessageGenerator(
+        ContentType.handshake,
+        bytearray(b'\x01\x00\x00\x00')))
+
+You can find example use of it in
+`test-aesccm.py <https://github.com/tomato42/tlsfuzzer/blob/master/scripts/test-aesccm.py>`_.
+
+Creating arbitrary messages
+---------------------------
+
+To send messages with a specific payload and type, while using encryption
+and record layer fragmentation, use
+:py:class:`~tlsfuzzer.messages.RawMessageGenerator`.
+
+It accepts two parameters that specify data sent to the other side
+(``content_type`` and ``data``) and one that stores message to print if
+processing of the message fails: ``description``.
+
+For example, to send an empty Finished message, write:
+
+.. code:: python
+
+    node = node.add_child(RawMessageGenerator(
+        ContentType.handshake,
+        bytearray(b'\x14\x00\x00\x00')
+
+You can find example use of it in
+`test-invalid-content-type.py <https://github.com/tomato42/tlsfuzzer/blob/master/scripts/test-invalid-content-type.py>`_.
+
+
+Modifying messages
+==================
+
+Tlsfuzzer supports applying two operations to sent messages: modifying
+length and modifying contents of specific bytes.
+
+Modifying length
+----------------
+
+Handshake messages include an internal header that identifies the message
+type and message length.
+Two methods can change their payload while modifying the header to match.
+
+The :py:func:`~tlsfuzzer.messages.pad_handshake` function adds data at the
+end of payload; ``size`` parameter specifies how many bytes and
+``pad_byte`` specifies the value of the added bytes.
```

Comment:
the ``pad_byte`` parameter

Reply:
technically it's both a param and a field of the object, but yes, the former use is more prevalent. fixed